coordinator/src/services/translation/request/prescribe/prescription-dispense.ts

```diff
@@ -17,72 +17,155 @@ import moment from "moment"
+    fhirHeader.sender,
+    fhirMedicationDispenses,
+    fhirFirstMedicationDispense
+  )
 
+  /* eslint-disable max-len */
   const hl7DispenseNotification = new hl7V3.DispenseNotification(new hl7V3.GlobalIdentifier(messageId))
-  hl7DispenseNotification.effectiveTime = convertMomentToHl7V3DateTime(moment.utc())
-
-  const fhirPatient = getPatientOrNull(bundle)
-  const hl7PatientId = getNhsNumber(fhirPatient, fhirFirstMedicationDispense)
-  const hl7Patient = createPatient(hl7PatientId)
+  hl7DispenseNotification.effectiveTime = hl7EffectiveTime
   hl7DispenseNotification.recordTarget = new hl7V3.DispenseRecordTarget(hl7Patient)
+  hl7DispenseNotification.primaryInformationRecipient = new hl7V3.PrimaryInformationRecipient(hl7AgentOrganization)
+  hl7DispenseNotification.pertinentInformation1 = hl7PertinentInformation1
+  hl7DispenseNotification.pertinentInformation2 = new hl7V3.DispensePertinentInformation2(hl7CareRecordElementCategory)
+  hl7DispenseNotification.sequelTo = new hl7V3.SequelTo(hl7PriorPrescriptionReleaseEventRef)
+  /* eslint-enable max-len */
 
-  const fhirHeader = getMessageHeader(bundle)
-  const fhirHeaderSender = fhirHeader.sender
-  const fhirHeaderDestination = onlyElement(fhirHeader.destination, "MessageHeader.destination")
+  return hl7DispenseNotification
+}
 
-  const hl7AgentOrganisationCode = fhirHeaderDestination.receiver.identifier.value
-  const hl7AgentOrganisationName = fhirHeaderDestination.receiver.display
-  const hl7AgentOrganization = createOrganisation(hl7AgentOrganisationCode, hl7AgentOrganisationName)
-  hl7DispenseNotification.primaryInformationRecipient = new hl7V3.PrimaryInformationRecipient()
-  hl7DispenseNotification.primaryInformationRecipient.AgentOrg = new hl7V3.AgentOrganization(hl7AgentOrganization)
+function createPertinentInformation1(
+  messageId: string,
+  fhirHeaderSender: fhir.IdentifierReference<fhir.Organization>,
+  fhirMedicationDispenses: Array<fhir.MedicationDispense>,
+  fhirFirstMedicationDispense: fhir.MedicationDispense
+): hl7V3.DispenseNotificationPertinentInformation1 {
 
   const fhirPractitionerPerformer = fhirFirstMedicationDispense.performer.find(p => p.actor.type === "Practitioner")
-  const hl7SupplyHeader = createSupplyHeader(
-    messageId,
-    fhirHeaderSender,
-    fhirMedicationDispenses,
-    fhirFirstMedicationDispense,
-    fhirPractitionerPerformer
+
+  const hl7RepresentedOrganisationCode = fhirHeaderSender.identifier.value
+  const hl7RepresentedOrganisationName = fhirHeaderSender.display
+  const hl7AuthorTime = fhirFirstMedicationDispense.whenPrepared
+  const hl7AgentPersonPersonName = fhirPractitionerPerformer.actor.display
+  const hl7PertinentPrescriptionStatus = createPertinentPrescriptionStatus(fhirFirstMedicationDispense)
+  const hl7PertinentPrescriptionIdentifier = createPertinentPrescriptionId(fhirFirstMedicationDispense)
+  const hl7PriorOriginalRef = createPriorOriginalRef(fhirFirstMedicationDispense)
+  const hl7Author = createAuthor(
```

Comment:
is there a way we can share this `createAuthor` function among the other translations? it just passes in strings (maybe the answer is no lol)

Reply:
Don't think so, implementation details are specific to dispense-notification